network/src/commonMain/kotlin/com/wire/kalium/network/KaliumKtorCustomLogging.kt

```diff
@@ -122,7 +127,7 @@ public class KaliumKtorCustomLogging private constructor(
         kaliumLogger.v("BODY Content-Type: $contentType")
         kaliumLogger.v("BODY START")
         val message = content.tryReadText(contentType?.charset() ?: Charsets.UTF_8) ?: "[response body omitted]"
-        kaliumLogger.v(message)
```

Comment:
isn't better to inject the obfuscation mechanism into the kaliumLogger and switch it on/off by configuration?

Reply:
then we can easily decide to obfuscate all the logs not by replacing each line of the logs.

Reply:
the thing is the logger module is not aware about the network stuff or Json elemnt and whatsoever , so I am not sure if we want to import these in the logger module just for this case

Reply:
> the thing is the logger module is not aware about the network stuff or Json elemnt and whatsoever , so I am not sure if we want to import these in the logger module just for this case

This would be a circular dependency.
`network -> logger -> network ERROR`

Reply:
`import kotlinx.serialization.decodeFromString
import kotlinx.serialization.json.Json
import kotlinx.serialization.json.JsonElement
import kotlinx.serialization.json.jsonArray
import kotlinx.serialization.json.jsonObject`

based on these lines, we need only the serialization library? I mean if we only add it to the logger that would be ok and no circular dependancy.
@vitorhugods @MohammedMokresh